### PR TITLE
Add docs for DetailsComponent

### PR DESCRIPTION
**Pull request by srt32**
image

### CODE REVIEW

**Comment by joelhawksley** [comment]
Screen Shot 2021-02-02 at 1 29 08 PM

🤔 not sure what's going on here, but at least on the Vercel preview, the examples here are not styled.

File: app/components/primer/details_component.rb [hunk 1]
```diff
@@ -1,11 +1,7 @@
 # frozen_string_literal: true
 
 module Primer
-  #
-  # overlay - options are `none`, `default` and `dark`. Dictates the type of overlay to render with.
-  # button - options are `default` and `reset`. default will make the target a default primer ``.btn`
-  #          reset will remove all styles from the <summary> element.
-  #
+  # Details classes are created to enhance the native behaviors of the details element
```
Comment (joelhawksley) [inline app/components/primer/details_component.rb:4]: How about:

```suggestion
  # Use DetailsComponent to reveal content after clicking a button.
```

Or something similar?

**Comment by srt32** [comment]
Interesting! It works as expected in chrome but not safari. I'm looking...

**Comment by srt32** [comment]
I am seeing "shadow content" in safari, which I assume is the problem here but I'm not sure why it's happening. Is that what we expect?

image

**Comment by joelhawksley** [comment]
> I am seeing "shadow content" in safari, which I assume is the problem here but I'm not sure why it's happening. Is that what we expect? 

I'm honestly not sure. @colebemis have you seen anything like this before?

**Comment by keithamus** [comment]
> I am seeing "shadow content" in safari, which I assume is the problem here but I'm not sure why it's happening. Is that what we expect?

That’s the browsers’ shadowdom implementation of the summary element. Nothing to worry about, just Safari devtools exposing their own internals. This isn’t related to the underlying issue, I believe.

**Comment by srt32** [comment]
b763ac1 removes the examples for now until we can resolve the safari rendering issues.

File: docs/content/components/timelineitem.md [hunk 2]
```diff
@@ -41,15 +41,21 @@ Use `TimelineItem` to display items on a vertical timeline, connected by badge e
 | `square` | `Boolean` | `true` | Whether to round the edges of the image. |
 | `system_arguments` | `Hash` | N/A | [System arguments](/system-arguments) |
 
+:nodoc
```
Comment (joelhawksley) [inline docs/content/components/timelineitem.md:44]: Might we omit these `nodocs` from the markdown output of the Rakefile?

Comment (srt32) [reply docs/content/components/timelineitem.md:44]: @manuelpuyol ^ thoughts?

Comment (srt32) [reply docs/content/components/timelineitem.md]: I resolved this problem for now by documenting the slots classes too ;)

**Comment by srt32** [comment]
Ready for one more :eye: here

File: app/components/primer/details_component.rb [hunk 3]
```diff
@@ -33,8 +32,10 @@ def render?
       summary.present? && body.present?
     end
 
-    # :nodoc
+    # Use the Summary slot  as the trigger button
```
Comment (joelhawksley) [inline app/components/primer/details_component.rb:35]:
```suggestion
    # Use the Summary slot as a trigger to reveal the content.
```

File: app/components/primer/details_component.rb [hunk 4]
```diff
@@ -33,8 +32,10 @@ def render?
...
     class Summary < Primer::Slot
+      # @param button [Boolean] If there should be a button or not
```
Comment (joelhawksley) [inline app/components/primer/details_component.rb:37]:
```suggestion
      # @param button [Boolean] Whether to render the Summary as a button or not.
```

File: app/components/primer/details_component.rb [hunk 5]
```diff
@@ -50,8 +51,9 @@ def component
       end
     end
 
-    # :nodoc
+    # Use the Body slot as the main content to be shown when triggered by the Summary
```
Comment (joelhawksley) [inline app/components/primer/details_component.rb:54]:
```suggestion
    # Use the Body slot as the main content to be shown when triggered by the Summary.
```

Comment (srt32) [reply app/components/primer/details_component.rb:54]: I was getting crashes including periods but I will try again!

**Review by keithamus — APPROVED** [review]
:+1: nice

**Pull request merged**